Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -1904,20 +1905,20 @@ public AndConstraint<TAssertions> BeUpperCased(string because = "", params objec
     public AndConstraint<TAssertions> NotBeUpperCased(string because = "", params object[] becauseArgs)
     {
         Execute.Assertion
-            .ForCondition(Subject is null || Subject.Any(ch => !char.IsUpper(ch)))
+            .ForCondition(Subject is null || HasMixedOrNoCase(Subject))
             .BecauseOf(because, becauseArgs)
-            .FailWith("Did not expect any characters in {context:string} to be upper cased{reason}.");
+            .FailWith("Expected some characters in {context:string} to be lower-case{reason}.");
```

Comment:
🤔 I'm pretty sure _all_ of the alphabetic characters should be lower case, right? Same in `BeLowerCased`

Reply:
No. The string `AbC` is not upper-cased, because if it was it would be `ABC`.